api/app/controllers/spree/api/address_book_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  module Api
+    class AddressBookController < Spree::Api::BaseController
+      # Note: the AddressBook is the resource to think about here, not individual addresses
+
+      before_filter :ensure_api_token
+
+      def mine
+         @user_addresses = current_api_user.user_addresses
+      end
+
+      def update
```

Comment:
If I'm deviating too far from "The Rails Way" I'm happy to have that discussion. I'm not creating an AddressBook in this action. Also, Addresses are readonly models, so they cannot be updated. This felt like the best compromise to me -- to think of the AddressBook as the resource.